ctru-rs/src/services/apt.rs

```diff
@@ -90,3 +132,48 @@ impl Drop for Apt {
         unsafe { ctru_sys::aptExit() };
     }
 }
+
+/// Can launch other applications when the current one exits.
+pub struct Chainloader<'a> {
+    _apt: &'a Apt,
+}
+
+impl<'a> Chainloader<'a> {
+    /// Gets a handle to the chainloader
+    pub fn new(apt: &'a Apt) -> Self {
```

Comment:
Maybe `Chainloader` should require a mutable reference to `Apt`, but that may not be necessary if *no* other aspect of `Apt` reads or uses data handled by `Chainloader`.

Tell me what you think.

Reply:
i tried this, but then using it inside the main `apt` loop is a mess because it takes an immutable reference to loop

Reply:
these are the only chainloader functions, so i think it's fine to have it be this way